src/celeritas/em/model/BraggModel.cc

```diff
@@ -0,0 +1,97 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/em/model/BraggModel.cc
+//---------------------------------------------------------------------------//
+#include "BraggModel.hh"
+
+#include "celeritas/Quantities.hh"
+#include "celeritas/em/data/BraggICRU73QOData.hh"
+#include "celeritas/em/executor/BraggICRU73QOExecutor.hh"
+#include "celeritas/em/interactor/detail/PhysicsConstants.hh"
+#include "celeritas/global/ActionLauncher.hh"
+#include "celeritas/global/CoreParams.hh"
+#include "celeritas/global/CoreState.hh"
+#include "celeritas/global/TrackExecutor.hh"
+#include "celeritas/phys/InteractionApplier.hh"
+#include "celeritas/phys/PDGNumber.hh"
+#include "celeritas/phys/ParticleParams.hh"
+#include "celeritas/phys/ParticleView.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct from model ID and other necessary data.
+ *
+ * TODO: This model also applies to hadrons.
+ */
+BraggModel::BraggModel(ActionId id, ParticleParams const& particles)
+    : ConcreteAction(id, "ioni-bragg", "interact by muon ionization (Bragg)")
+{
+    CELER_EXPECT(id);
+
+    data_.inc_particle = particles.find(pdg::mu_plus());
+    data_.electron = particles.find(pdg::electron());
+
+    CELER_VALIDATE(data_.electron && data_.inc_particle,
+                   << "missing electron and/or mu+ particles (required for "
+                   << this->description() << ")");
+
+    data_.electron_mass = particles.get(data_.electron).mass();
+    data_.lowest_kin_energy = detail::bragg_lowest_kin_energy();
+
+    CELER_ENSURE(data_);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Particle types and energy ranges that this model applies to.
+ */
+auto BraggModel::applicability() const -> SetApplicability
+{
+    Applicability applic;
+    applic.particle = data_.inc_particle;
+    applic.lower = zero_quantity();
```

Comment:
should this be  `applic.lower = data_.lowest_kin_energy`;?

Reply:
I don't think so, since it's not the model energy limit, just used in the interactor to calculate the minimum delta ray energy. I think these models should be valid at all energies below 200 keV.

Reply:
Okay, then the comment,

>  //! Kinetic energy limit for the Bragg model 

 Is somewhat misleading.  Probably, should it be `the kinetic energy limit of the delta ray` or something related to that (the minimum energy transfer from muon to electron?)?  We know that the process is valid down to 100eV which is the lower limit of the lambda and dE/dx, but how can the muon below the limit can transfer the kinetic energy above the limit to generate a delta ray? Anyway, I agree that it is not the lower limit of the model, which I confused from the comment - same for the ICRU model.

Reply:
That's a good point, I'll update the documentation as you suggested. The minimum energy transfer from the incident particle to the delta ray is the smaller of the cutoff energy and `kin_energy_limit * inc_mass / proton_mass`, so I don't thiink we can use the latter as the lower model energy limit since it could still be larger than the cutoff.